httpx/utils.py

```diff
@@ -42,6 +42,8 @@ def str_query_param(value: typing.Optional[typing.Union[str, int, float, bool]])
         return "false"
     elif value is None:
         return ""
+    elif isinstance(value, bytes):
```

Comment:
The signature for `value` isn't `PrimitiveData`, shame mypy didn't catch this.

Reply:
Probably because both actually refer to the same type, just with a different name. I updated the code so we use `PrimitiveData` everywhere. :)

Reply:
Oh I thought `PrimitiveData` included bytes (hence your isinstance() call). Do we want to include bytes here?

Reply:
Ah I see! I also thought PrimitiveData included bytes from the beginning, but it never did. So this is technically a new feature then (bytes support in query params) — probably worth adding right? If that’s reasonable I’ll see if it makes sense to add bytes to PrimitiveData, or include bytes as an extra option here as you initially mentioned.

Reply:
It seems reasonable to me. :)